### PR TITLE
luci-proto-wireguard: fix value validation

### PR DESCRIPTION
**Pull request by feckert**
* add warning that `allowed_ips` must not be empty

* verify last base64 string symbol is an `=` sig
If this is not the case I get the following error during connection
```
Thu Apr 30 11:37:42 2020 daemon.notice netifd: wg1 (23647): Key is not the correct length or format: `3UW+WgCasdu1SEOngdwYZm2jc5ogAQx9GQAjCt520g01'
Thu Apr 30 11:37:42 2020 daemon.notice netifd: wg1 (23647): Configuration parsing error
```

### CODE REVIEW

File: protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ function validateBase64(section_id, value) {
 	if (value.length != 44 || !value.match(/^(?:[A-Za-z0-9+\/]{4})*(?:[A-Za-z0-9+\/]{2}==|[A-Za-z0-9+\/]{3}=)?$/))
 		return _('Invalid Base64 key string');
 
+	if (value['43'] != "=" )
+		return _('Invalid Base64 key string');
```
Comment (jow-) [inline protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js:14]: That looks wrong. `=` is a padding char, not every valid base64 string ends with a `=` char.

Comment (feckert) [reply protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js:14]: I did a `wg genkey` and I have not seen any other char then `=`  so I thought this is a must?
Maybe wireguard use this base64 diffrence? We could ask @zx2c4 if he could give us a hint.

Comment (jow-) [reply protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js:14]: Seems you're right, apparently a trailing `=` is mandatory:
https://github.com/WireGuard/wireguard-tools/blob/3bf1b64d44fdbe589ac145b61bd4b3aa62b625bf/src/encoding.c#L56

So fine with me then, but please use a numeric subscript (`value[43]` vs. `value['43']`)

**Comment by feckert** [comment]
@jow- thanks for your review

**Comment by zx2c4** [comment]
If you guys really want to nerd out, match against this regex:

```^[A-Za-z0-9+/]{42}[A|E|I|M|Q|U|Y|c|g|k|o|s|w|4|8|0]{1}=$```

This ensures that the key is actually 256-bits and not 257 or 258 bits.

**Pull request merged**